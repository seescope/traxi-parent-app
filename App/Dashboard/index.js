import React from 'react';
import lodash from 'lodash';
import { connect } from 'react-redux';
import { ScrollView, } from 'react-native';
import * as Animatable from 'react-native-animatable';
import Swiper from 'react-native-swiper';

import Spacing from '../Components/Spacing';
import { VERY_LIGHT_GREY } from '../Constants/Colours';
import KidCircle from './Components/KidCircle';
import Card from './Components/Card';
import TopApp from './Components/TopApp';
import TopCategory from './Components/TopCategory';
import PeakTime from './Components/PeakTime';
import RecentApp from './Components/RecentApp';

const containerStyle = {
  alignItems: 'center',
  paddingVertical: 36,
  paddingHorizontal: 8,
  backgroundColor: VERY_LIGHT_GREY,
};

const outerContainerStyle = {
  backgroundColor: VERY_LIGHT_GREY,
};

const getTodayUsage = peakTimes => ((peakTimes && peakTimes.week)
    ? (lodash.find(peakTimes.week, { name: 'Today'}) || { usage: 0}).usage
    : 0);

const DashboardScreen = ({ loading, kid, topApps, topCategories, peakTimes, recentApps }) =>
  <ScrollView style={outerContainerStyle} contentContainerStyle={containerStyle}>
    <Animatable.View
      useNativeDriver
      animation="bounceIn"
    >
      <KidCircle loading={loading} kid={kid} usage={getTodayUsage(peakTimes)} />
    </Animatable.View>

    <Spacing height={32} />

    <Card loading={loading} header="Top Apps" Component={TopApp} data={topApps} />

    <Card loading={loading} header="Top Categories" Component={TopCategory} data={topCategories} />

    <Card loading={loading} header="Peak Times" Component={PeakTime} data={peakTimes} />

    <Card loading={loading} header="Recent Apps" Component={RecentApp} data={recentApps} />
  </ScrollView>

DashboardScreen.propTypes = {
  topApps: React.PropTypes.object,
  topCategories: React.PropTypes.object,
  peakTimes: React.PropTypes.object,
  kid: React.PropTypes.object,
  recentApps: React.PropTypes.array,
  loading: React.PropTypes.bool.isRequired,
};

const mapStateToProps = ({ reports, kids, loading }) => ({
  kids,
  reports,
  loading,
});

const Dashboard = ({ reports, kids, loading }) =>
  <Swiper>
    {kids.map(kid => <DashboardScreen key={kid.UUID} loading={loading} kid={kid} {...reports[kid.UUID]} />)}
  </Swiper>

Dashboard.propTypes = {
  reports: React.PropTypes.object.isRequired,
  kids: React.PropTypes.array.isRequired,
  loading: React.PropTypes.bool.isRequired,
};

export default connect(mapStateToProps)(Dashboard);
