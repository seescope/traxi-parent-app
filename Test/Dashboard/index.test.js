import React from 'react';
import Dashboard from '../../App/Dashboard';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

it('renders correctly', () => {
  const mockStore = configureStore();
  const testStore = mockStore({ 
    loading: true,
    kids: [
      {
        UUID: 'abc-123',
      },
    ],
    reports: {
      'abc-123': {
        topApps: {},
        topCategories: {},
        peakTimes: {},
        recentApps: [],
      },
    },
  });

  const tree = renderer.create(<Provider store={testStore}><Dashboard /></Provider>);
  expect(tree.toJSON()).toMatchSnapshot();
});
