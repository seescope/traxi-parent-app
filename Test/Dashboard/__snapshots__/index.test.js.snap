// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
        "position": "relative",
      },
      Object {
        "height": 1334,
        "width": 750,
      },
    ]
  }
>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    autoplay={false}
    autoplayDirection={true}
    autoplayTimeout={2.5}
    bounces={false}
    contentContainerStyle={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    contentOffset={Object {}}
    horizontal={true}
    index={0}
    loadMinimal={false}
    loadMinimalSize={1}
    loop={true}
    onMomentumScrollEnd={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    pagingEnabled={true}
    removeClippedSubviews={true}
    scrollsToTop={false}
    showsButtons={false}
    showsHorizontalScrollIndicator={false}
    showsPagination={true}
    showsVerticalScrollIndicator={false}
  >
    <View>
      <View
        style={
          Array [
            Object {
              "height": 1334,
              "width": 750,
            },
            Object {
              "backgroundColor": "transparent",
            },
          ]
        }
      >
        <RCTScrollView
          contentContainerStyle={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F0F2F7",
              "paddingHorizontal": 8,
              "paddingVertical": 36,
            }
          }
          style={
            Object {
              "backgroundColor": "#F0F2F7",
            }
          }
        >
          <View>
            <View
              animation="bounceIn"
              delay={500}
              useNativeDriver={true}
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#545454",
                      "fontSize": 32,
                      "fontWeight": "200",
                      "marginBottom": 16,
                    }
                  }
                />
                <View />
                <Image
                  source={
                    Object {
                      "uri": undefined,
                    }
                  }
                  style={
                    Object {
                      "borderRadius": 359,
                      "height": 718,
                      "marginBottom": -718,
                      "top": -728,
                      "width": 718,
                    }
                  }
                />
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "rgba(0, 0, 0, 0.45)",
                      "borderRadius": 359,
                      "height": 718,
                      "justifyContent": "center",
                      "marginBottom": -718,
                      "top": -728,
                      "width": 718,
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#FFF",
                        "fontSize": 96,
                        "fontWeight": "100",
                      }
                    }
                  >
                    0
                  </Text>
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#FFF",
                        "fontSize": 19,
                        "fontWeight": "200",
                      }
                    }
                  >
                    minutes
                     online today
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "height": 32,
                }
              }
            />
            <View
              style={
                Object {
                  "backgroundColor": "#FFF",
                  "borderRadius": 4,
                  "margin": 8,
                  "marginBottom": 24,
                  "padding": 8,
                  "shadowColor": "#000000",
                  "shadowOffset": Object {
                    "height": 3,
                    "width": 3,
                  },
                  "shadowOpacity": 0.16,
                  "shadowRadius": 3,
                  "width": 734,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#545454",
                    "fontSize": 19,
                  }
                }
              >
                Top Apps
              </Text>
              <View
                style={
                  Object {
                    "backgroundColor": "#A6A8AB",
                    "height": 1,
                    "marginBottom": 24,
                    "marginTop": 8,
                    "width": 56,
                  }
                }
              />
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "marginBottom": 16,
                  }
                }
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "flex": 1,
                      "height": 24,
                      "opacity": 1,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#448AFF",
                        "flex": 1,
                        "fontSize": 15,
                        "fontWeight": "400",
                        "textAlign": "center",
                      }
                    }
                  >
                    Today
                  </Text>
                </View>
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "flex": 1,
                      "height": 24,
                      "opacity": 1,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "flex": 1,
                        "fontSize": 15,
                        "fontWeight": "200",
                        "textAlign": "center",
                      }
                    }
                  >
                    Yesterday
                  </Text>
                </View>
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "flex": 1,
                      "height": 24,
                      "opacity": 1,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "flex": 1,
                        "fontSize": 15,
                        "fontWeight": "200",
                        "textAlign": "center",
                      }
                    }
                  >
                    Last 7 Days
                  </Text>
                </View>
              </View>
              <View>
                <View>
                  <ActivityIndicator
                    animating={true}
                    color="#448AFF"
                    hidesWhenStopped={true}
                    size="large"
                    style={
                      Object {
                        "marginBottom": 16,
                      }
                    }
                  />
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        undefined,
                        Object {
                          "backgroundColor": "transparent",
                          "color": "#545454",
                          "textAlign": "center",
                        },
                      ]
                    }
                  >
                    Loading...
                  </Text>
                </View>
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                isTVSelectable={true}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  style={
                    Object {
                      "alignItems": "flex-end",
                      "flex": 1,
                      "height": 32,
                      "justifyContent": "flex-end",
                    }
                  }
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "backgroundColor": "#FFF",
                  "borderRadius": 4,
                  "margin": 8,
                  "marginBottom": 24,
                  "padding": 8,
                  "shadowColor": "#000000",
                  "shadowOffset": Object {
                    "height": 3,
                    "width": 3,
                  },
                  "shadowOpacity": 0.16,
                  "shadowRadius": 3,
                  "width": 734,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#545454",
                    "fontSize": 19,
                  }
                }
              >
                Top Categories
              </Text>
              <View
                style={
                  Object {
                    "backgroundColor": "#A6A8AB",
                    "height": 1,
                    "marginBottom": 24,
                    "marginTop": 8,
                    "width": 56,
                  }
                }
              />
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "marginBottom": 16,
                  }
                }
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "flex": 1,
                      "height": 24,
                      "opacity": 1,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#448AFF",
                        "flex": 1,
                        "fontSize": 15,
                        "fontWeight": "400",
                        "textAlign": "center",
                      }
                    }
                  >
                    Today
                  </Text>
                </View>
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "flex": 1,
                      "height": 24,
                      "opacity": 1,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "flex": 1,
                        "fontSize": 15,
                        "fontWeight": "200",
                        "textAlign": "center",
                      }
                    }
                  >
                    Yesterday
                  </Text>
                </View>
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "flex": 1,
                      "height": 24,
                      "opacity": 1,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "flex": 1,
                        "fontSize": 15,
                        "fontWeight": "200",
                        "textAlign": "center",
                      }
                    }
                  >
                    Last 7 Days
                  </Text>
                </View>
              </View>
              <View>
                <View>
                  <ActivityIndicator
                    animating={true}
                    color="#448AFF"
                    hidesWhenStopped={true}
                    size="large"
                    style={
                      Object {
                        "marginBottom": 16,
                      }
                    }
                  />
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        undefined,
                        Object {
                          "backgroundColor": "transparent",
                          "color": "#545454",
                          "textAlign": "center",
                        },
                      ]
                    }
                  >
                    Loading...
                  </Text>
                </View>
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                isTVSelectable={true}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  style={
                    Object {
                      "alignItems": "flex-end",
                      "flex": 1,
                      "height": 32,
                      "justifyContent": "flex-end",
                    }
                  }
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "backgroundColor": "#FFF",
                  "borderRadius": 4,
                  "margin": 8,
                  "marginBottom": 24,
                  "padding": 8,
                  "shadowColor": "#000000",
                  "shadowOffset": Object {
                    "height": 3,
                    "width": 3,
                  },
                  "shadowOpacity": 0.16,
                  "shadowRadius": 3,
                  "width": 734,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#545454",
                    "fontSize": 19,
                  }
                }
              >
                Peak Times
              </Text>
              <View
                style={
                  Object {
                    "backgroundColor": "#A6A8AB",
                    "height": 1,
                    "marginBottom": 24,
                    "marginTop": 8,
                    "width": 56,
                  }
                }
              />
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "marginBottom": 16,
                  }
                }
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "flex": 1,
                      "height": 24,
                      "opacity": 1,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "color": "#448AFF",
                        "flex": 1,
                        "fontSize": 15,
                        "fontWeight": "400",
                        "textAlign": "center",
                      }
                    }
                  >
                    Today
                  </Text>
                </View>
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "flex": 1,
                      "height": 24,
                      "opacity": 1,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "flex": 1,
                        "fontSize": 15,
                        "fontWeight": "200",
                        "textAlign": "center",
                      }
                    }
                  >
                    Yesterday
                  </Text>
                </View>
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "flex": 1,
                      "height": 24,
                      "opacity": 1,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "flex": 1,
                        "fontSize": 15,
                        "fontWeight": "200",
                        "textAlign": "center",
                      }
                    }
                  >
                    Last 7 Days
                  </Text>
                </View>
              </View>
              <View>
                <View>
                  <ActivityIndicator
                    animating={true}
                    color="#448AFF"
                    hidesWhenStopped={true}
                    size="large"
                    style={
                      Object {
                        "marginBottom": 16,
                      }
                    }
                  />
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        undefined,
                        Object {
                          "backgroundColor": "transparent",
                          "color": "#545454",
                          "textAlign": "center",
                        },
                      ]
                    }
                  >
                    Loading...
                  </Text>
                </View>
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                isTVSelectable={true}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  style={
                    Object {
                      "alignItems": "flex-end",
                      "flex": 1,
                      "height": 32,
                      "justifyContent": "flex-end",
                    }
                  }
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "backgroundColor": "#FFF",
                  "borderRadius": 4,
                  "margin": 8,
                  "marginBottom": 24,
                  "padding": 8,
                  "shadowColor": "#000000",
                  "shadowOffset": Object {
                    "height": 3,
                    "width": 3,
                  },
                  "shadowOpacity": 0.16,
                  "shadowRadius": 3,
                  "width": 734,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#545454",
                    "fontSize": 19,
                  }
                }
              >
                Recent Apps
              </Text>
              <View
                style={
                  Object {
                    "backgroundColor": "#A6A8AB",
                    "height": 1,
                    "marginBottom": 24,
                    "marginTop": 8,
                    "width": 56,
                  }
                }
              />
              <View>
                <View>
                  <ActivityIndicator
                    animating={true}
                    color="#448AFF"
                    hidesWhenStopped={true}
                    size="large"
                    style={
                      Object {
                        "marginBottom": 16,
                      }
                    }
                  />
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        undefined,
                        Object {
                          "backgroundColor": "transparent",
                          "color": "#545454",
                          "textAlign": "center",
                        },
                      ]
                    }
                  >
                    Loading...
                  </Text>
                </View>
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hitSlop={undefined}
                isTVSelectable={true}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  style={
                    Object {
                      "alignItems": "flex-end",
                      "flex": 1,
                      "height": 32,
                      "justifyContent": "flex-end",
                    }
                  }
                />
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
